my_methods.rb

```diff
@@ -0,0 +1,182 @@
+        my_each { |val| return false unless val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false unless val.is_a?(var) }
+      else
+        my_each { |val| return false if val != var }
+      end
+      return true
+    end
+    unless block_given?
+      my_each { |val| return false if val == false || val.nil? }
+      return true
+    end
+    my_each do |num|
+      return false unless yield(num)
+    end
+    true
+  end
+
+  def my_any?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return true if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return true if val.is_a?(var) }
+      else
+        my_each { |val| return true if val == var }
+      end
+      return false
+    end
+    unless block_given?
+      my_each { |val| return true if val != false && !val.nil? }
+      return false
+    end
+
+    my_each do |num|
+      return true if yield(num)
+    end
+    false
+  end
+
+  def my_none?(var = nil)
+    unless var.nil?
+      if var.is_a?(Regexp)
+        my_each { |val| return false if val.match(var) }
+      elsif var.is_a?(Module)
+        my_each { |val| return false if val.is_a?(var) }
+      else
+        my_each { |val| return false if val == var }
+      end
+      return true
+    end
+    unless block_given?
+      my_each { |val| return false if val != false && !val.nil? }
+      return true
+    end
+
+    my_each do |num|
+      return false if yield(num)
+    end
+    true
+  end
+
+  def my_count(var = nil)
+    i = 0
+    unless var.nil?
+      my_each { |val| i += 1 if val == var }
+      return i
+    end
+    return size unless block_given?
+
+    i = 0
+    my_each do |num|
+      i += 1 if yield(num)
+    end
+    i
+  end
+
+  def my_map(var = nil)
+    arr = []
+    if var.is_a?(Proc)
+      if is_a?(Array)
+        my_each { |num| arr << var.call(num) }
+      elsif is_a?(Hash)
+        my_each { |_key, _val| arr << var.call(num) }
+      end
+      return arr
+    end
+
+    return to_enum(:my_map) unless block_given?
+
+    if is_a?(Array)
+      my_each { |num| arr << yield(num) }
+    elsif is_a?(Hash)
+      my_each { |key, val| arr << yield(key, val) }
+    end
+    arr
+  end
+
+  def my_inject(in1 = nil, in2 = nil)
+    arr = !is_a?(Array) ? to_a.flatten : flatten
+    res = in1.nil? || !in1.is_a?(Numeric) ? 0 : in1
+    return to_enum(:my_inject) if in1.nil? && in2.nil? && !block_given?
+
+    if in1.is_a?(Symbol)
+      operator = in1
+    elsif in2.is_a?(Symbol)
+      operator = in2
+    end
+
+    if arr.my_all?(String)
+      str = ''
+      arr.my_each { |val| str = yield(str, val) }
+      return str
+    end
+
+    arr.my_each { |num| return arr unless num.is_a?(Numeric) }
+
+    unless operator.nil?
+      arr.my_each { |num| res = res.send(operator, num) }
+      return res
+    end
+
+    return to_enum(:my_inject) unless block_given?
+
+    arr.my_each { |num| res = yield(res, num) }
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving an array instead of the result
![image](https://user-images.githubusercontent.com/52209067/80552199-15578600-8983-11ea-8845-6d8a1b2bdb64.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)